### PR TITLE
Import Feature/oidf 8 json url encoder

### PR DESCRIPTION
**Pull request by sanderPostma**

### CODE REVIEW

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoder.kt [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+private val qpAllowedChars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_.~".toSet()
+
+/**
+ * URL encode a String.
+ * Converts characters not allowed in URL query parameters to their percent-encoded equivalents.
+ *
+ * @return URL encoded String
+ */
+fun urlEncodeValue(input: String): String {
```
Comment (nklomp) [inline modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoder.kt:16]: Please anotate the function with @jsexport

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoder.kt [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    return Json.encodeToString(serializer, this).toUrlEncodedValue()
+}
+
+/**
+ * Extension function to decode a URL encoded String.
+ * Converts percent-encoded characters back to their original form.
+ *
+ * @return Decoded String
+ */
+fun String.fromUrlEncodedValue(): String {
```
Comment (nklomp) [inline modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoder.kt:76]: Please create a regular function with @JSexport first, then call that from this extension function. This allows the most flexibility in deciding whether we want to make the extension functions available to javascript for instance, or whether to only expose the regular functions

**Review by nklomp — CHANGES_REQUESTED** [review]
Some small remarks on jsexport and creating a regular function first before using them as extension function

File: modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoderTest.kt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+@Serializable
+data class TestData(
+    val grants: Grants,
+    val credential_configuration_ids: List<String>,
+    val credential_issuer: String
+)
+
+class JsonUrlEncoderTest {
+    private val originalJson =
+        """{"grants":{"urn:ietf:params:oauth:grant-type:pre-authorized_code":{"pre-authorized_code":"a"}},"credential_configuration_ids":["Woonplaatsverklaring"],"credential_issuer":"https://agent.issuer.bd.demo.sphereon.com"}"""
```
Comment (jcmelati) [inline modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoderTest.kt:25]: We should always use dummy data when creating tests. Please remove all references to other projects.

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoder.kt (hunk 1 above)
Comment (sanderPostma) [reply modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoder.kt:16]: Fixed in next commit

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoder.kt (hunk 2 above)
Comment (sanderPostma) [reply modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoder.kt:76]: Fixed in next commit

File: modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoderTest.kt (hunk 3 above)
Comment (sanderPostma) [reply modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoderTest.kt:25]: Fixed in next commit

**Pull request merged**